src/main/java/sirius/db/es/ElasticQuery.java

```diff
@@ -1465,18 +1468,33 @@ public void truncate() {
      * <p>
      * Use this for larger result sets where integrity and constraints do not matter or are managed manually.
      *
-     * @throws OptimisticLockException if one of the documents was modified during the runtime of the truncate
+     * @throws OptimisticLockException               if one of the documents was modified during the runtime of the truncate
+     * @throws sirius.kernel.health.HandledException if the {@linkplain LowLevelClient#deleteByQuery(String, String, JSONObject) deleteByQuery}
+     *                                               request aborted due to any unrecoverable errors during the process
      */
     public void tryTruncate() throws OptimisticLockException {
         if (forceFail) {
             return;
         }
 
         String filteredRouting = checkRouting(Elastic.RoutingAccessMode.WRITE);
-        elastic.getLowLevelClient()
-               .deleteByQuery(computeEffectiveIndexName(elastic::determineWriteAlias),
-                              filteredRouting,
-                              buildSimplePayload());
+        JSONObject deleteByQueryResponse = elastic.getLowLevelClient()
+                                                  .deleteByQuery(computeEffectiveIndexName(elastic::determineWriteAlias),
+                                                                 filteredRouting,
+                                                                 buildSimplePayload());
+        if (Boolean.TRUE.equals(deleteByQueryResponse.getBoolean(KEY_TIMED_OUT))
+            || deleteByQueryResponse.getIntValue(KEY_VERSION_CONFLICTS) > 0) {
+            Elastic.LOG.WARN("Truncate timed out or had version conflicts:\n" + deleteByQueryResponse.toJSONString());
+            throw new OptimisticLockException();
+        }
+        if (deleteByQueryResponse.getJSONArray(KEY_FAILURES) != null
+            && !deleteByQueryResponse.getJSONArray(KEY_FAILURES).isEmpty()) {
+            Elastic.LOG.SEVERE("Truncate aborted due to unrecoverable error(s):\n"
+                               + deleteByQueryResponse.toJSONString());
+            throw Exceptions.createHandled()
+                            .withSystemErrorMessage("Truncate aborted due to unrecoverable error(s)!")
+                            .handle();
+        }
```

Comment:
shouldn't we switch this (move above OL warning)...

Reply:
I think not.

> If `version_conflicts` is filled, those conflicts will also be listed in `failures`.

Version conflicts can be resolved by retrying the deletion for the conflicting documents in a new request, which is done by the automatic retry when throwing and catching the OLE. If we move the check for `failures` before the OLE and `failures` contains only version conflicts, we lose the option to retry. See https://discuss.elastic.co/t/version-conflict-while-using-delete-by-query/104831 for a sample response containing `version_conflicts` in `failures`.